gatsby-node.js

```diff
@@ -401,21 +411,64 @@ const getLearnPageFilters = async () => {
     return filters;
 };
 
-exports.onCreatePage = async ({ page, actions }) => {
-    if (page.path === '/learn/') {
-        const { createPage, deletePage } = actions;
-        const allArticles = await getLearnPageArticles();
-        const featuredArticles = getFeaturedLearnArticles(allArticles);
-        const filters = await getLearnPageFilters(allArticles);
-        deletePage(page);
-        createPage({
-            ...page,
-            context: {
-                ...page.context,
-                allArticles,
-                featuredArticles,
-                filters,
-            },
-        });
+const getFeaturedArticlesForPage = async (
+    page,
+    defaultFeaturedArticles,
+    allArticles = null
+) => {
+    if (!allArticles) {
+        allArticles = await getLearnPageArticles();
+    }
+    // TODO: Verify field names with docsp
+    const featuredArticleSlugs =
+        getNestedValue(['featuredArticles', `${page}Page`], metadata) ||
+        defaultFeaturedArticles;
+    const featuredArticles = getFeaturedArticles(
+        allArticles,
+        featuredArticleSlugs
+    );
+    return featuredArticles;
+};
+
+exports.onCreatePage = async ({
+    page,
+    actions: { createPage, deletePage },
+}) => {
+    switch (page.path) {
+        case '/learn/':
+            const allArticles = await getLearnPageArticles();
```

Comment:
Is it worth looking into caching this?

Reply:
I'm wondering if we should be doing this work is the same way we create the other dynamic pages. Instead of having a 'static' page in `pages` they would be templates.

With or without that change I wonder if there's an opportunity to ensure we're not making the same stitch calls more than once. We could store a map of returned calls and check there before reaching out to the server.

All this is done at build time, so I don't think it's a blocking issue (unless the build time becomes untenable), but this file is starting to get unwieldy and there's room for some cleanup and organization (and tests) now.

Reply:
Ah yes we can mimic our own cache here like that, sounds good!

I also agree on the need for cleanup. I was thinking we could move some of the lengthier API methods out (which would also allow them to be more testable). Now seems like a great time to do so, though, since more of the incoming features will use this file.